lib/KeyStorage.php

```diff
@@ -208,30 +200,37 @@ protected function deleteUsersPublicKey(string $uid): void {
 	 *
 	 * @throws NotPermittedException
 	 */
-	protected function deleteUsersPrivateKey(string $uid): void {
+	protected function deleteUsersPrivateKey(ISimpleFolder $privateKeysRoot, string $uid): void {
 		$fileName = $this->getFileNameForPrivateKey($uid);
 
 		try {
-			$this->appData->getFolder($this->privateKeysRoot)
-				->getFile($fileName)
-				->delete();
-		} catch (NotFoundException $e) {
+			$privateKeysRoot->getFile($fileName)->delete();
+		} catch (NotFoundException) {
 			return;
 		}
 	}
 
 	/**
 	 * @throws NotFoundException
 	 * @throws NotPermittedException
+	 * @return array{privateKeysRoot: ISimpleFolder, publicKeysRoot: ISimpleFolder}
 	 */
-	protected function verifyFolderStructure(): void {
-		$keyStorageRoot = $this->appData->getFolder('/');
-		if (!$keyStorageRoot->fileExists($this->privateKeysRoot)) {
-			$this->appData->newFolder($this->privateKeysRoot);
+	protected function verifyFolderStructure(): array {
+		try {
+			$privateKeysRoot = $this->appData->getFolder($this->privateKeysRoot);
```

Comment:
Should this be cached in the object to avoid refetching?
```suggestion
			$this->privateKeysRoot = $this->appData->getFolder($this->privateKeysRoot);
```